plugins/processors/gpuattributes/processor_test.go

```diff
@@ -19,93 +19,170 @@ func TestProcessMetrics(t *testing.T) {
-				"ClusterName": "cluster",
-				"Drop":        "val",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+			metrics: generateMetrics("node", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"Drop":        "val",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 1,
-			want: map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\"}",
+			want: []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\"}",
+				},
 			},
 		},
 		"dropPodWithoutPodName": {
-			metrics: generateMetrics("pod", map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+			metrics: generateMetrics("pod", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 0,
-			want:    map[string]string{},
+			want:    []map[string]string{},
 		},
-		"keepPodWithoutPodName": {
-			metrics: generateMetrics("pod", map[string]string{
-				"ClusterName": "cluster",
-				"PodName":     "pod",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+		"keepPodWithPodName": {
+			metrics: generateMetrics("pod", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"PodName":     "pod",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 1,
-			want: map[string]string{
-				"ClusterName": "cluster",
-				"PodName":     "pod",
-				"kubernetes":  "{\"host\":\"test\"}",
+			want: []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"PodName":     "pod",
+					"kubernetes":  "{\"host\":\"test\"}",
+				},
 			},
 		},
 		"dropContainerWithoutPodName": {
-			metrics: generateMetrics("container", map[string]string{
-				"ClusterName": "cluster",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+			metrics: generateMetrics("container", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 0,
-			want:    map[string]string{},
+			want:    []map[string]string{},
 		},
-		"keepContainerWithoutPodName": {
-			metrics: generateMetrics("container", map[string]string{
-				"ClusterName": "cluster",
-				"PodName":     "pod",
-				"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+		"keepContainerWithPodName": {
+			metrics: generateMetrics("container", []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"PodName":     "pod",
+					"kubernetes":  "{\"host\":\"test\",\"b\":\"2\"}",
+				},
 			}),
 			wantCnt: 1,
-			want: map[string]string{
-				"ClusterName": "cluster",
-				"PodName":     "pod",
-				"kubernetes":  "{\"host\":\"test\"}",
+			want: []map[string]string{
+				{
+					"ClusterName": "cluster",
+					"PodName":     "pod",
+					"kubernetes":  "{\"host\":\"test\"}",
+				},
+			},
+		},
+		"dropSingleDatapointWithoutPodName": {
```

Comment:
Just curious, where are the names: "dropSingleDatapointWithoutPodName", coming from?

Reply:
they are just test names